### PR TITLE
feat: document ratelimit headers

### LINKED ISSUE #3: document ratelimit headers

### PR DESCRIPTION
**Pull request by EnokiUN**
closes #3

### CODE REVIEW

File: src/models/ratelimits.md [hunk 1]
```diff
@@ -75,6 +75,42 @@ be requested from the CDN, as well as maximum filesize limits.
...
+>
+> These headers also exist on `429` responses.
+>
+> <hr />
+>
+> The UNIX timestamps here all use the actual UNIX epoch and not the Eludris one.
+
+### How To Implement Rate Limiting
+
+Ideally how rate limiting is implementing is that you check the response headers
```
Comment (Sharp-Eyes) [inline src/models/ratelimits.md:103]: > Ideally how rate limiting is implementing is that you check the response headers

implemented*, probably

File: src/models/ratelimits.md [hunk 2]
```diff
@@ -75,6 +75,42 @@ be requested from the CDN, as well as maximum filesize limits.
...
+on every request and check if you've ran out of requests for this interval.
+
+If so you prevent any other requests on the same bucket for `Now - (X-RateLimit-Last-Reset + X=RateLimit+Reset)`
```
Comment (Sharp-Eyes) [inline src/models/ratelimits.md:106]: I'd add a comma after "If so";

> `X=RateLimit+Reset`

nice

File: src/models/ratelimits.md [hunk 3]
```diff
@@ -75,6 +75,42 @@ be requested from the CDN, as well as maximum filesize limits.
...
+milliseconds.
+
+Additionally when dealing with Effis you should also check if the uploaded files
```
Comment (Sharp-Eyes) [inline src/models/ratelimits.md:109]: > Additionally when dealing with Effis you should [...]

I'd make this "Additionally, when dealing with Effis, you should [...]" or "Additionally - when dealing with Effis - you should [...]"

File: src/models/ratelimits.md [hunk 4]
```diff
@@ -75,6 +75,42 @@ be requested from the CDN, as well as maximum filesize limits.
...
+abide by the limits of file sizes in the [InstanceInfo](./instance_info.md) payload,
+you should also keep track of how many bytes left you can send and limit requests
+to not surpass that limit.
+
 ## Relevant Endpoints
```
Comment (Sharp-Eyes) [inline src/models/ratelimits.md:114]: possibly add `/info` as a relevant endpoint

File: src/models/ratelimits.md [hunk 5]
```diff
@@ -75,6 +75,42 @@ be requested from the CDN, as well as maximum filesize limits.
...
+Ideally how rate limiting is implementing is that you check the response headers
+on every request and check if you've ran out of requests for this interval.
+
+If so you prevent any other requests on the same bucket for `Now - (X-RateLimit-Last-Reset + X=RateLimit+Reset)`
+milliseconds.
+
+Additionally when dealing with Effis you should also check if the uploaded files
+abide by the limits of file sizes in the [InstanceInfo](./instance_info.md) payload,
+you should also keep track of how many bytes left you can send and limit requests
+to not surpass that limit.
```
Comment (Sharp-Eyes) [inline src/models/ratelimits.md:112]: I'd put a full stop after payload;

> you should also keep track of how many bytes left you can send

Maybe "You should also keep track of how many bytes you have left to send" sounds a little better? Not entirely sure on that one.

**Review by Sharp-Eyes — COMMENTED** [review]
*sips tea*

File: src/models/ratelimits.md (hunk 1 above)
Comment (Sharp-Eyes) [reply src/models/ratelimits.md:103]: Maybe even "Ideally, rate limiting is implemented such that you check the [...]"?

Comment (EnokiUN) [reply src/models/ratelimits.md:103]: I think I'm actually against that, it just sounds weird as opposed to how simple it currently is (with the implemented fix)

File: src/models/rate_limits.md [hunk 6]
```diff
@@ -100,19 +100,21 @@ supposed to use these to properly rate limit any clients or API wrapper you make
 
 ### How To Implement Rate Limiting
 
-Ideally how rate limiting is implementing is that you check the response headers
+Ideally how rate limiting is implemented is that you check the response headers
 on every request and check if you've ran out of requests for this interval.
 
-If so you prevent any other requests on the same bucket for `Now - (X-RateLimit-Last-Reset + X=RateLimit+Reset)`
+If so, you prevent any other requests on the same bucket for `Now - (X-RateLimit-Last-Reset + X-RateLimit+Reset)`
```
Comment (Sharp-Eyes) [inline src/models/rate_limits.md]: > X-RateLimit+Reset

nice 2: electric boogaloo.

**Review by Sharp-Eyes — APPROVED** [review]
i approb

**Pull request merged**